### DISCUSSION TITLE
List of details to polish

### DISCUSSION THREAD

**Discussion by Bouh** [4ian/GDevelop#2355] [Archive]
- [x]  ~~In the new instruction editor, **Create an object** is only available after a choice of an object.
This action is generic and should be in both lists, but before all available by the search bar without choose object or non-object list. _Basically the research bar shouldn't be tab dependent._~~

- [ ] The console show errors types, it's benefic for dev, we must show them the errors. 
`Error: "You must enter a number or an valid expression."`
But it's not visible in the eventsheet, Only operators can be highlighted in red.
![image](https://user-images.githubusercontent.com/1670670/97712127-50677d00-1abe-11eb-9755-07359dd86140.png)
Even if the engine manage the wrong values and return the default zero, an user can spend time to research the error because it's not the value desired.

Accessorily I was thinking of having an error manager.
Which would list:
```
Type errors
Useless files
Objects not existing but present in the event sheet.
Events of any type. (Unknown or unsupported instruction)
Possible after deleting an extension for example.
Missing resources (like audio files)
```

- [x] The auto updater have trouble?
When users are on an old beta the auto updated dosen't work even is the option has been enabled back.
Users on beta 99 or older, can't update GD. The auto updater said there is no new version.

- [x] The export on a older version of GD can not working with the current build service.
Users on b99 can't export online, and there is no message for alert the users for update the software to the lastest.

**Comment by 4ian** [comment]
> In the new instruction editor, Create an object is only available after a choice of an object.
> This action is generic and should be in both lists, but before all available by the search bar without choose object or non-object list.

Agree that this action should probably be shown in both cases, it's not intuitive in the current case.

> Basically the research bar shouldn't be tab dependent.

It is already independent in the sense that whatever the tab, the search is always done on objects and on "free" actions/conditions. But not on object/behaviors "actions/conditions". We could add it as supplementary results but that would ask to choose an object first... 
But maybe it's not a real problem. Once you get to know that you need to select an object first to find something like "rotate", you won't search "rotate" again in the first screen. But some actions like "Create an object" are at the "frontier" between object and not object, so could be in both.

> But it's not visible in the eventsheet, Only operators can be highlighted in red.
> Even if the engine manage the wrong values and return the default zero, an user can spend time to research the error because it's not the value desired.

I agree it's a bad usability problem. The issue is that we can't run the parser to check for errors for each parameter, it would consume too much resources and slow down the UI a lot.
BUT as you mentioned, we could have an error manager. 

It would work as a "store of errors", and when exporting it would be passed to the exporter/code generation functions. These code generation functions would report errors to this instead of just writing to the console. We could then show them in the UI when launching a preview. Better we could only show the "new errors" since the last preview :) 
Finally, we could read this store of errors and use it to display errors in the events sheet in a fairly efficient way.

> The auto updater have trouble?

Mmm not sure what happened, I don't remember updating electron-builder or the dependencies that take care of auto-update.. :/
We could maybe have an API endpoint to get the latest version, so that we could at least say "auto-update might not be working, but there is a new version so please download it!".

**Comment by 4ian** [comment]
@Bouh Fixing the Create an object issue is probably just reworking a bit these objects:
https://github.com/4ian/GDevelop/blob/98c9763d1cf0b578b6e91d8777b0c40b183169f9/newIDE/app/src/InstructionOrExpression/EnumerateInstructions.js#L20-L61

For the "error manager", let's call it the Error Report (or Diagnostics Report?) and make a card for it in the roadmap: "Add a Diagnostic report to warn about potential issues in the game and show them in the events sheet".

**Comment by Bouh** [comment]
- Create an object is done in #2073

- Trello card for Diagnostics Report is [here](https://trello.com/c/IiLgNR16/462-add-a-diagnostic-report-to-warn-about-potential-issues-in-the-game-and-show-them-in-the-events-sheet).

**Comment by AlexandreSi** [comment]
Locking this discussion since it is discussed in https://github.com/4ian/GDevelop/discussions/2919.
